### PR TITLE
editoast: refactor rollings stocks, livery, electrical profiles tests

### PR DESCRIPTION
**Pull request by Wadjetz**
part of [#6980](https://github.com/OpenRailAssociation/osrd/issues/6980)

Refactor of tests for:

- [x] Rolling stock
- [x] Rolling stock livery
- [x] Electrical profiles

### CODE REVIEW

**Review by hamz2a — APPROVED** [review]

File: editoast/src/modelsv2/fixtures.rs [hunk 1]
```diff
@@ -45,6 +52,119 @@ pub async fn create_study(conn: &mut DbConnection, name: &str, project_id: i64)
         .expect("Failed to create study")
 }
 
+pub fn fast_rolling_stock_form(name: &str) -> RollingStockForm {
```
Comment (woshilapin) [inline editoast/src/modelsv2/fixtures.rs:55]: I believe some of the functions in this file don't have to be `pub`.

File: editoast/src/modelsv2/rolling_stock_model.rs [hunk 2]
```diff
@@ -223,98 +218,72 @@ impl From<RollingStock> for RollingStockModelChangeset {
...
-    async fn update_rolling_stock(db_pool: Arc<DbConnectionPool>) {
-        use crate::modelsv2::Update;
-        let mut db_conn = db_pool.get().await.expect("Failed to get db connection");
         // GIVEN
-        let other_rs_name = "other_rolling_stock_update_rolling_stock";
-        let rolling_stock =
-            named_fast_rolling_stock("fast_rolling_stock_update_rolling_stock", db_pool.clone())
-                .await;
-        let rolling_stock_id = rolling_stock.id();
+        let rs_name_with_enrgy_sources_name = "other_rolling_stock_update_rolling_stock";
```
Comment (woshilapin) [inline editoast/src/modelsv2/rolling_stock_model.rs]:
```suggestion
        let rs_name_with_energy_sources_name = "other_rolling_stock_update_rolling_stock";
```

File: editoast/src/modelsv2/rolling_stock_model.rs [hunk 3]
```diff
@@ -223,98 +218,72 @@ impl From<RollingStock> for RollingStockModelChangeset {
...
 
-        let other_rolling_stock_id = other_rolling_stock.id();
-        let mut other_rolling_stock =
-            RollingStockModel::retrieve(&mut db_conn, other_rolling_stock_id)
-                .await
-                .unwrap()
-                .unwrap();
-        other_rolling_stock.name = name.to_string();
+        // Creating the second rolling stock
+        let rs_name_with_enrgy_sources_name = "fast_rolling_stock_with_energy_sources_name";
```
Comment (woshilapin) [inline editoast/src/modelsv2/rolling_stock_model.rs]:
```suggestion
        let rs_name_with_energy_sources_name = "fast_rolling_stock_with_energy_sources_name";
```

File: editoast/src/views/light_rolling_stocks.rs [hunk 4]
```diff
@@ -119,137 +118,145 @@ async fn get_by_name(
...
         let uri = format!("/light_rolling_stock/?page_size={count}");
-        let req = TestRequest::get().uri(&uri).to_request();
-        let response = call_service(&app, req).await;
+        let request = TestRequest::get().uri(&uri).to_request();
+        let response = call_service(&app.service, request).await;
         let response: PaginatedResponse<LightRollingStockWithLiveries> =
             assert_status_and_read!(response, StatusCode::OK);
 
-        // Ensure that AT LEAST all the rolling stocks create above are returned, in order
+        //  Ensure that AT LEAST all the rolling stocks create above are returned, in order
```
Comment (woshilapin) [inline editoast/src/views/light_rolling_stocks.rs]: Interesting change 😄

File: editoast/src/views/rolling_stocks/mod.rs [hunk 5]
```diff
@@ -581,429 +572,492 @@ async fn create_compound_image(
...
+        TestRequest::get()
+            .uri(format!("/rolling_stock/{rolling_stock_id}").as_str())
+            .to_request()
+    }
+
+    pub fn rolling_stock_delete_request(rolling_stock_id: i64) -> Request {
+        TestRequest::delete()
+            .uri(format!("/rolling_stock/{rolling_stock_id}").as_str())
+            .to_request()
+    }
```
Comment (woshilapin) [inline editoast/src/views/rolling_stocks/mod.rs]: These functions probably don't need to be `pub`? (I didn't check if they're used somewhere else but I would find it a weird pattern for a `mod tests` to expose an API).

File: editoast/src/views/rolling_stocks/mod.rs [hunk 6]
```diff
@@ -581,429 +572,492 @@ async fn create_compound_image(
...
+        assert!(power_restrictions.contains(&"C2".to_string()));
+        assert!(power_restrictions.contains(&"C5".to_string()));
+        let response_body: Vec<String> = read_body_json(response).await;
+        assert!(response_body.contains(&"C2".to_string()));
+        assert!(response_body.contains(&"C5".to_string()));
     }
 
     #[rstest]
-    async fn update_locked_successfully(db_pool: Arc<DbConnectionPool>) {
+    #[ignore] // TODO test for rolling stock with DBConnectionPoolV2
```
Comment (woshilapin) [inline editoast/src/views/rolling_stocks/mod.rs:1001]: Is it planned to be follow-up in a future PR, or is it something you forgot to address on the current PR?
(note: there is a few others TODO below too).

**Review by woshilapin — APPROVED** [review]
Huge work on the tests, thank you 🙏 Only minor comments.

File: editoast/src/views/rolling_stocks/mod.rs (hunk 6 above)
Comment (Wadjetz) [reply editoast/src/views/rolling_stocks/mod.rs:1001]: It using the train schedule V1, with @leovalais we decided to wait the removal of train schedule V1 to finish

File: editoast/src/modelsv2/fixtures.rs (hunk 1 above)
Comment (Wadjetz) [reply editoast/src/modelsv2/fixtures.rs:55]: This function is used in some tests

**Pull request merged**